### PR TITLE
ci: Run the tests against Symfony6

### PR DESCRIPTION
**Pull request by theofidry**
This is mostly an extract from #1775.

After this PR what should be left is adding the complementary jobs to the CI (lowest/highest).

### CODE REVIEW

**Comment by theofidry** [comment]
I really can't tell if the STTY issue is really related or not

File: devTools/phpstan-src.neon [hunk 1]
```diff
@@ -47,11 +47,15 @@ parameters:
             message: '#^Function shell_exec is unsafe to use#'
             path: '../src/TestFramework/CommandLineBuilder.php'
 
+
+
```
Comment (maks-rafalko) [inline devTools/phpstan-src.neon:51]: not needed?

**Review by maks-rafalko — COMMENTED** [review]
what is the issue with STTY?

**Comment by theofidry** [comment]
>what is the issue with STTY?

no longer appearing... Not sure why...

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -21,8 +21,7 @@ jobs:
         # TODO: include highest & lowest
         dependencies: [ locked ]
         coverage-driver: [ pcov, xdebug ]
-        # TODO: include Symfony6 later
-        symfony-version: [ '5.4.*' ]
+        symfony-version: [ '5.4.*', '6.*.*' ]
```
Comment (maks-rafalko) [inline .github/workflows/tests.yaml:24]: not sure I understand how it works now.

All the builds have now `locked` value for `dependencies` key. Each GH job runs `composer install`, ignoring (as I see) SF 5.4.* or 6..* - jobs are just use the versions from the lock file.

- should we add `highest` and `lowest` on line 21?
- should we vary SF versions **only** when `dependencies` is not `locked`?

Comment (theofidry) [reply .github/workflows/tests.yaml:24]: Resolved in #1835

**Comment by theofidry** [comment]
It would be best to exclude the combination `locked & 6.*.*` but not sure how to do that

**Comment by sidz** [comment]
@theofidry see https://github.com/infection/infection/pull/1775/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR27 if you need to exclude smth

**Comment by theofidry** [comment]
@sidz I'll do that in a separate PR I think; it's only optimizations after all

**Review by maks-rafalko — APPROVED** [review]

**Pull request merged**